packages/hardhat-test-helpers/src/helpers/setStorageAt.ts

```diff
@@ -0,0 +1,31 @@
+import type { NumberLike } from "../types";
+import {
+  getHardhatProvider,
+  assertValidAddress,
+  assertHexString,
+  toRpcQuantity,
+} from "../utils";
+
+export async function setStorageAt(
+  hexAddress: string,
+  index: NumberLike,
+  code: string
```

Comment:
We should accept more types of values here. Requiring a string of 32 bytes is not very ergonomic.

I think accepting a `NumberLike` would be enough. Then after converting it to a hex-encoded version, we should pad it so that it has 32 bytes, as it's required by the spec for this method.